### ISSUE TITLE
Edx Oauth2 provider support

### ISSUE THREAD

**Issue by dominicfollett** [simplabs/rails_api_auth#41]
I've added Oauth2 provider support, for the popular Edx learning platform. Would you be interested in looking at a pull request? :) Or is this perhaps out of the spirit of what you want to provide out of the box?

**Comment by marcoow** [comment]
You mean auth code flow for Edx? That would probably be fine to add.

**Comment by dominicfollett** [comment]
Yes :) I'll submit a pull request as soon as I have written some tests.
